crates/hyperqueue/src/server/event/log/write.rs

```diff
@@ -52,14 +66,26 @@ impl EventLogWriter {
         Ok(())
     }
 
-    async fn write_buffer(&mut self) -> tokio::io::Result<()> {
-        self.file.write_all(&self.buffer).await?;
+    async fn write_buffer(&mut self) -> anyhow::Result<()> {
+        let mut buffer_gzipped =
+            GzipEncoder::with_quality(Vec::with_capacity(CHUNK_MAX_SIZE * 2), Level::Fastest);
+
+        let _ = buffer_gzipped.write_all(&self.buffer).await?;
+        let _ = buffer_gzipped.flush().await;
```

Comment:
This `flush` is not needed and just adds extra data by the compression codec. `shutdown`'s documentation claims that it does `flush` implicitly.